### PR TITLE
[quantization] add a pass to convert fake_quant annotated IR

### PR DESCRIPTION
**Pull request by wyzero**
An example is shown as following.

convert:
```
 const -> fake_quant
                     \
 input -> fake_quant -> gemm -> fake_quant ->
```
to
```
 const -> quantize
                    \
 input -> quantize -> quantized_gemm -> dequantize ->
```

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+    rewriter.replaceOp(op, dequantizedOut);
+    return success();
+  }
+};
+
+void populateQuantizedPatterns(RewritePatternSet& patterns) {
+  // clang-format off
+  patterns.insert<
+    QuantizedDynamicConvOpConverter
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc:118]: So there will be a quntized gemm converter? I see the document is demonstrating GEMM.

File: tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    if (!convOp) return failure();
+
+    auto inputFakeQuantOp =
+        convOp.lhs().getDefiningOp<mhlo_disc::FakeQuantOp>();
+    auto weightFakeQuantOp =
+        convOp.rhs().getDefiningOp<mhlo_disc::FakeQuantOp>();
+    if (!inputFakeQuantOp || !weightFakeQuantOp) return failure();
+    if (inputFakeQuantOp.use_signed() != weightFakeQuantOp.use_signed() ||
+        inputFakeQuantOp.num_bits() != weightFakeQuantOp.num_bits()) {
+      return failure();
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc:61]: nit: maybe some error message?

**Review by qiuxiafei — COMMENTED** [review]
on going...

File: tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc (hunk 1 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc:118]: Yes. I leave the quantized GEMM for you ^v^~

File: tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc (hunk 2 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/disc_convert_fake_quant_op.cc:61]: It's just not supported by this pattern, not an error though.

**Review by qiuxiafei — APPROVED** [review]
LGTM

**Pull request merged**